### PR TITLE
Implement chart/manifest merging

### LINKED ISSUE #1976: Zarf chart / manifest composability merging
### Is your feature request related to a problem? Please describe.
Given components and manifests can be composed together with importable components and that the manifest and chart names need to be unique, it would be nice to merge chart and manifest definitions that share the same name together so that you could add values files or manifests into the existing helm chart or manifest.

### Describe the solution you'd like

- **Given** I have a base component
- **And** a component that I want to import into
- **When** I add a `chart` or `manifest` that shares a name with the base component's `charts` or `manifests`
- **Then** Zarf merges the `valuesFiles` or `files` keys along with the `namespace` key (if it is not empty) for that `chart` or `manifest`

### Describe alternatives you've considered
This could be done more holistically but that may get too far into Helm logic and this should provide enough customizablility along with more clarity of these names needing to be unique.

### Additional context
This should also ensure that manifest and chart names are in fact unique during package validation.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This implements more intelligent merging functionality based on the uniqueness of names in `charts` and `manifests`

## Related Issue

Fixes #1976

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Review by cmwylie19 — CHANGES_REQUESTED** [review]
It doesn't work in practice

**Review by Noxsios — CHANGES_REQUESTED** [review]
small tweaks, overall logic looked solid

File: src/pkg/utils/yaml.go [hunk 1]
```diff
@@ -85,15 +86,21 @@ func ColorPrintYAML(data any, hints map[string]string, spaceRootLists bool) {
...
 
 	// Inject the hints into the colorized YAML
 	for key, value := range hints {
-		colorizedYAML = strings.Replace(colorizedYAML, key, value, 1)
+		outputYAML = strings.Replace(outputYAML, key, value, 1)
+	}
+
+	if config.NoColor {
+		// If no color is specified strip any color codes from the output
+		ansiRegex := regexp.MustCompile(`\x1b\[(.*?)m`)
```
[the remark this answers is not in the document]
Comment (Racer159) [reply src/pkg/utils/yaml.go:98]: Done!

**Review by Noxsios — APPROVED** [review]

**Review by cmwylie19 — APPROVED** [review]
lgtm

**Pull request merged**